app/src/main/java/org/mozilla/fenix/components/accounts/FenixAccountManager.kt

```diff
@@ -0,0 +1,92 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.accounts
+
+import android.content.Context
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.lifecycleScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import mozilla.components.browser.menu.BrowserMenuBuilder
+import mozilla.components.browser.menu.BrowserMenuItem
+import mozilla.components.concept.sync.AccountObserver
+import mozilla.components.concept.sync.AuthType
+import mozilla.components.concept.sync.OAuthAccount
+import org.mozilla.fenix.ext.components
+
+/**
+ * TODO
+ *
+ */
+open class FenixAccountManager(
+    private val context: Context,
+    private val lifecycleOwner: LifecycleOwner
+) {
+    val accountManager = context.components.backgroundServices.accountManager
+    val authenticatedAccount = accountManager.authenticatedAccount() != null
```

Comment:
This should probably be `isAccountAuthenticated` to make it clear it's a boolean variable rather than the actual authenticated account.